packages/shared/src/retry.ts

```diff
@@ -0,0 +1,108 @@
+type Milliseconds = number;
+
+type RetryOptions = Partial<{
+  /**
+   * The initial delay before the first retry.
+   * @default 125
+   */
+  firstDelay: Milliseconds;
+  /**
+   * The maximum delay between retries.
+   * If set to 0, the delay will increase indefinitely.
+   * @default 0
+   */
+  maxDelay: Milliseconds;
+  /**
+   * The multiplier for the exponential backoff.
+   * @default 2
+   */
+  factor: number;
```

Comment:
How about `backoffFactor` or something similar?

Reply:
Similarly to my comment about `jitter`, `factor` is the most commonly used term for this and I think adding `backoff` doesn't add much in this case